File: Backend/tecnpoint/src/main/java/com/techsolutions/tecnpoint/entities/Conversa.java
package com.techsolutions.tecnpoint.entities;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;
import java.time.temporal.ChronoUnit;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Builder
@Table(name = "conversa")
public class Conversa {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id_conversa")
    private Long idConversa;

    @Column(name = "mensagem", columnDefinition = "TEXT")
    private String mensagem;

    @ManyToOne
    @JoinColumn(name = "fk_id_chamado", nullable = false)
    private Chamados chamado;

    @ManyToOne
    @JoinColumn(name = "fk_id_remetente", nullable = false)
    private Usuarios remetente;

    @Column(name = "data_hora", insertable = false, updatable = false,
            columnDefinition = "TIMESTAMP DEFAULT CURRENT_TIMESTAMP")
    private LocalDateTime dataHoraEnvio;

    @PrePersist
    protected void onCreate(){
        // Fallback caso o banco não aplique o DEFAULT
        if (dataHoraEnvio == null) {
            dataHoraEnvio = LocalDateTime.now();
        }
    }
}

File: Desktop/TecnPoint/TecnPoint.Modelos/Conversa.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace TecnPoint.Modelos
{
    public class Conversa
    {
        private long _idConversa { get; set; }
        private string _mensagem { get; set; }
        private Chamado _chamado { get; set; }
        private Usuario _remetente { get; set; }
        private DateTime _dataHoraEnvio { get; set; }

        public Conversa() { }

        public long IdConversa
        {
            get { return _idConversa; }
            set { _idConversa = value; }
        }

        public string Mensagem
        {
            get { return _mensagem; }
            set { _mensagem = value; }
        }

        public Chamado Chamado
        {
            get { return _chamado; }
            set { _chamado = value; }
        }

        public Usuario Remetente
        {
            get { return _remetente; }
            set { _remetente = value; }
        }

        public DateTime DataHoraEnvio
        {
            get { return _dataHoraEnvio; }
            set { _dataHoraEnvio = value; }
        }
    }
}
